crates/polars-time/src/upsample.rs

```diff
@@ -119,15 +119,52 @@ fn upsample_impl(
     stable: bool,
 ) -> PolarsResult<DataFrame> {
     let s = source.column(index_column)?;
+    if offset.parsed_int || every.parsed_int {
+        polars_ensure!(
+            ((offset.parsed_int || offset.is_zero())
+             && (every.parsed_int || every.is_zero())),
+            ComputeError: "you cannot combine time durations like '2h' with integer durations like '3i'"
+        )
+    }
     ensure_sorted_arg(s, "upsample")?;
-    if matches!(s.dtype(), DataType::Date) {
+    let time_type = s.dtype();
+    if matches!(time_type, DataType::Date) {
         let mut df = source.clone();
-        df.try_apply(index_column, |s| {
+        df.apply(index_column, |s| {
             s.cast(&DataType::Datetime(TimeUnit::Milliseconds, None))
+                .unwrap()
+        })
+        .unwrap();
+        let mut out = upsample_impl(&df, by, index_column, every, offset, stable).unwrap();
```

Comment:
Could you propagate this error instead of unwrapping it? so `let mut out = upsample_impl(&df, by, index_column, every, offset, stable)?;`

With casting to Datetime above, we know it's going to work - casting Date to Datetime(Milliseconds) will always work - but `upsample_impl` could presumably still fail, right?

Reply:
I agree. Fixed them.